xrpl/ledger-entry-types/common_fields.go

```diff
@@ -0,0 +1,17 @@
+package ledger
+
+import "github.com/Peersyst/xrpl-go/xrpl/transaction/types"
+
+// Every entry in a ledger's state data has the same set of common fields, plus additional fields based on the ledger entry type.
+// Field names are case-sensitive.
+type LedgerEntryCommonFields struct {
+	// The unique ID for this ledger entry.
+	// In JSON, this field is represented with different names depending on the context and API method.
+	// (Note, even though this is specified as "optional" in the code, every ledger entry should have one unless it's legacy data from very early in the XRP Ledger's history.)
+	Index       types.Hash256 `json:"index,omitempty"`
```

Comment:
I think it would be nice to have only one of them, since having two add more complexity afterwards (I would keep `LedgerIndex` since its more descriptive imo)

Reply:
I saw that xrpl.js has `Index`. Maybe we can keep `Index` for now as the doc says it's generally `Index`:
```
Generally, a ledger entry's ID is returned as the index field in JSON, at the same level as the object's contents.
```